__tests__/api.test.js

```diff
@@ -2,90 +2,134 @@ const app = require("../app.js");
+            expect(article).toHaveProperty("topic", expect.any(String));
+            expect(article).toHaveProperty("created_at", expect.any(String));
+            expect(article).toHaveProperty("votes", expect.any(Number));
+            expect(article).toHaveProperty(
+              "article_img_url",
+              expect.any(String)
+            );
+            expect(article).toHaveProperty("comment_count", expect.any(String)); // I thought this would be a number
+          });
+        });
+    });
+  });
+  describe("/api/articles/:article_id/comments", () => {
+    test("sends all comments to the user based on an article_id", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then((response) => {
+          const { comments }  = response.body
+          expect(comments).toBeSortedBy('created_at', { descending: true });
+          comments.forEach((comment) => {
+            expect(comment).toHaveProperty("comment_id", expect.any(Number));
+            expect(comment).toHaveProperty("votes", expect.any(Number));
+            expect(comment).toHaveProperty("created_at", expect.any(String));
+            expect(comment).toHaveProperty("author", expect.any(String));
+            expect(comment).toHaveProperty("body", expect.any(String));
+            expect(comment).toHaveProperty("article_id", expect.any(Number));
+          });
+        });
+    });
+    test("Sends an appropriate status and error message when given an invalid id", () => {
+      return request(app)
+        .get("/api/articles/999/comments")
+        .expect(404)
+        .then((response) => {
+          expect(response.body.msg).toBe("article does not exist");
+        });
+    });
+  });
+  test("Sends an appropriate status and error message when provided invalid data type on comments", () => {
+    return request(app)
+      .get("/api/articles/ninety/comments")
+      .expect(400)
       .then((response) => {
-        const { articles } = response.body;
-        expect(articles.length).toBeGreaterThan(0);
-          articles.forEach((article)=>{ 
-            expect(article).toHaveProperty("author", expect.any(String))
-            expect(article).toHaveProperty("title", expect.any(String))
-            expect(article).toHaveProperty("article_id", expect.any(Number))
-            expect(article).not.toHaveProperty("body", expect.any(String))
-            expect(article).toHaveProperty("topic", expect.any(String))
-            expect(article).toHaveProperty("created_at", expect.any(String))
-            expect(article).toHaveProperty("votes", expect.any(Number))
-            expect(article).toHaveProperty("article_img_url", expect.any(String))
-            expect(article).toHaveProperty("comment_count", expect.any(String)) // I thought this would be a number
-          })
-            
+        expect(response.body.msg).toBe("Bad request");
       });
-  })
-})})
\ No newline at end of file
+  });
+});
```

Comment:
Good testing here! There's one more that you can do. 

There's an article that does exist but has no comments yet (37). In that case we would want an empty array back and a 200 because it is a valid request. In the future it may have comments added.